http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/listener/http2/HTTP2SourceHandler.java

```diff
@@ -0,0 +1,310 @@
+        cMsg.setProperty(Constants.HTTP_VERSION, Constants.HTTP2_VERSION);
+        cMsg.setProperty(org.wso2.carbon.messaging.Constants.LISTENER_PORT,
+                ((InetSocketAddress) ctx.channel().localAddress()).getPort());
+        cMsg.setProperty(org.wso2.carbon.messaging.Constants.LISTENER_INTERFACE_ID, listenerConfiguration.getId());
+        cMsg.setProperty(org.wso2.carbon.messaging.Constants.PROTOCOL, Constants.PROTOCOL_NAME);
+        if (listenerConfiguration.getSslConfig() != null) {
+            cMsg.setProperty(Constants.IS_SECURED_CONNECTION, true);
+        } else {
+            cMsg.setProperty(Constants.IS_SECURED_CONNECTION, false);
+        }
+        cMsg.setProperty(Constants.LOCAL_ADDRESS, ctx.channel().localAddress());
+        cMsg.setProperty(Constants.LOCAL_NAME, ((InetSocketAddress) ctx.channel().localAddress()).getHostName());
+        cMsg.setProperty(Constants.REMOTE_ADDRESS, ctx.channel().remoteAddress());
+        cMsg.setProperty(Constants.REMOTE_HOST, ((InetSocketAddress) ctx.channel().remoteAddress()).getHostName());
+        cMsg.setProperty(Constants.REMOTE_PORT, ((InetSocketAddress) ctx.channel().remoteAddress()).getPort());
+        ChannelHandler handler = ctx.handler();
+        cMsg.setProperty(Constants.CHANNEL_ID, ((HTTP2SourceHandler) handler).getListenerConfiguration().getId());
+        cMsg.setProperty(Constants.STREAM_ID, streamId);
+        if (headers.path() != null) {
+            String path = headers.getAndRemove(Constants.HTTP2_PATH).toString();
+            cMsg.setProperty(Constants.TO, path);
+            cMsg.setProperty(Constants.REQUEST_URL, path);
+        }
+        if (headers.method() != null) {
+            String method = headers.getAndRemove(Constants.HTTP2_METHOD).toString();
+            cMsg.setProperty(Constants.HTTP_METHOD, method);
+        }
+        // Remove PseudoHeaderNames from headers
+        headers.getAndRemove(Constants.HTTP2_AUTHORITY);
+        headers.getAndRemove(Constants.HTTP2_SCHEME);
+
+        // Copy Http2 headers to carbon message
+        headers.forEach(k -> cMsg.setHeader(k.getKey().toString(), k.getValue().toString()));
+        streamIdRequestMap.put(streamId, cMsg);
+        return cMsg;
+    }
+
+    @Override
+    public void onPriorityRead(ChannelHandlerContext ctx, int streamId, int streamDependency,
+                               short weight, boolean exclusive) {
+    }
+
+    @Override
+    public void onRstStreamRead(ChannelHandlerContext ctx, int streamId, long errorCode) {
+    }
+
+    @Override
+    public void onSettingsAckRead(ChannelHandlerContext ctx) {
+    }
+
+    @Override
+    public void onSettingsRead(ChannelHandlerContext ctx, Http2Settings settings) {
+    }
+
+    @Override
+    public void onPingRead(ChannelHandlerContext ctx, ByteBuf data) {
+    }
+
+    @Override
+    public void onPingAckRead(ChannelHandlerContext ctx, ByteBuf data) {
+    }
```

Comment:
IMO We can not throw exception as these methods provides to do any customization based on frame type. As an example when upgrade done client send setting ack frame and we do not do nothing. If user cancel download it will send rst frame and server do not need to do anything.